pkg/apis/batch/interfaces.go

```diff
@@ -0,0 +1,29 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package batch
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+)
+
+func (obj *Job) PodSpec() (*api.PodSpec, string) {
```

Comment:
> I would think external types should have this equivalence logic as well.

Right, if this were on external types would it be ok?  If so, can I add both, along with reflective tests to ensure we don't slip and we'll have something to use now and something to use later?